um_run.py

```diff
@@ -0,0 +1,315 @@
+                path_flags=[
+                    AddFlags("$source/jules/*", ['-DUM_JULES']),
+                    AddFlags("$source/um/*", ['-I', '$relative/include']),
+
+                    # coupling defines
+                    AddFlags("$source/um/control/timer/*", ['-DC97_3A']),
+                    AddFlags("$source/um/io_services/client/stash/*", ['-DC96_1C']),
+                ],
+            ),
+
+            Analyse(
+                root_symbol='um_main',
+                unreferenced_deps=None,
+
+                # fparser2 fails to parse this file, but it does compile.
+                special_measure_analysis_results=[
+                    AnalysedFile(
+                        fpath=Path(
+                            os.path.expanduser("~/git/fab/tmp-workspace/um_atmos_safe/build_output/casim/lookup.f90")),
+                        file_hash=None,
+                        symbol_defs=['lookup'],
+                        symbol_deps=['mphys_die', 'variable_precision', 'mphys_switches', 'mphys_parameters', 'special',
+                                     'passive_fields', 'casim_moments_mod', 'yomhook', 'parkind1'],
+                        file_deps=[],
+                        mo_commented_file_deps=[]),
+                ]
+            ),
+
+            CompileC(compiler='gcc', common_flags=['-c', '-std=c99']),
+
+            CompileFortran(
+                compiler=os.path.expanduser('~/.conda/envs/sci-fab/bin/gfortran'),
+                common_flags=[
+                    '-fdefault-integer-8', '-fdefault-real-8', '-fdefault-double-8',
+                    '-c',
+                    '-J', '$output',  # .mod file output and include folder
+                ],
+
+                path_flags=[
+                    # mpl include - todo: just add this for everything?
+                    AddFlags("$output/um/*", ['-I', os.path.expanduser("~/git/fab/tmp-workspace/gcom/build_output")]),
+                    AddFlags("$output/jules/*",
+                             ['-I', os.path.expanduser("~/git/fab/tmp-workspace/gcom/build_output")]),
+
+                    # todo: allow multiple filters per instance?
+                    *[AddFlags(*i) for i in ALLOW_MISMATCH_FLAGS]
+                ]
+            ),
+
+            # todo: ArchiveObjects(),
+
+            #
+            LinkExe(
+                # linker='gcc',
+                linker=os.path.expanduser('~/.conda/envs/sci-fab/bin/mpifort'),
+                flags=[
+                    '-lc', '-lgfortran', '-L', '~/.conda/envs/sci-fab/lib',
+                    '-L', os.path.expanduser('~/git/fab/tmp-workspace/gcom/build_output'), '-l', 'gcom'
+                ],
+                output_fpath='$output/../um_atmos.exe')
+        ],
+
+    )
+
+
+# TODO: REVIEWERS, SHOULD WE NEED THESE?
+# a not recommended flag?
+# todo: allow a list of filters?
+ALLOW_MISMATCH_FLAGS = [
+    ('*/hardware_topology_mod.f90', ['-fallow-argument-mismatch']),
```

Comment:
Might've stored the actual flag text in a variable - what if the syntax for the mismatch flag changes just slightly at the next compiler version?  (it happens, e.g. Intel changed their openMP flag from `-fopenmp` to `-qopenmp` at one stage)